ui/src/app/base/hooks.js

```diff
@@ -305,3 +305,54 @@ export const useOnWindowResize = (callback) => {
     };
   }, [callback]);
 };
+
+/**
+ * Handle checking when a value has cycled from false to true.
+ * @param {boolean} value - The value to check.
+ * @param {() => void} onCycled - The function to call when the value changes from false to true.
```

Comment:
Is this really the correct jsdoc string or is this a find and replace error?

Reply:
Yeah this is TypeScript syntax for JSDoc annotations. We could also use Closure syntax (`function(): void`) but I like the idea of maintaining the syntax you see in type definitions.

Reply:
Weird!

Reply:
@huwshimi hahaha not a fan? I could just use the basic `function` param in the meantime, maybe it would be better to get consensus on it first

Reply:
Nah it's fine I just didn't know what it was. It's strange how it's using code to document the code.